tests/pamiq_core/trainer/test_base/test_base.py

```diff
@@ -0,0 +1,118 @@
+from abc import ABC, abstractmethod
+from typing import Any
+
+import pytest
+
+from pamiq_core.data import DataUser, DataUsersDict
+from pamiq_core.model import (
+    TrainingModel,
+    TrainingModelsDict,
+)
+
+from ._dummy_impls import (
+    DummyDataBuffer,
+    DummyInferenceModel,
+    DummyTrainer,
+    DummyTrainingModel,
+)
+
+
+class TestTrainer:
```

Comment:
@zassou65535 抽象メソッドのテストも追加
```suggestion
class TestTrainer:
    def test_abstractmethod(self):
        assert Trainer.__abstractmethods__ == frozenset({"train"})
```